src/Manipulator/ObjectManipulator.php

```diff
@@ -35,10 +35,15 @@ public static function addInstance(
         $parentAssociationMappings = $parentFieldDescription->getParentAssociationMappings();
```

Comment:
Shouldnt this be typed instead with all the options for the array? (or at least the known options)

Reply:
Kinda related to https://github.com/sonata-project/SonataAdminBundle/issues/6274.

We're using `fieldName` key (mandatory) and `mappedBy` (optionnal) in SonataAdmin.
But a lot more are used in persistence bundle.

Only adding those field to the array would be too limited. Adding all the persistence bundle ones would be too much.
Best would be to use a value object.

Reply:
Btw `array{fieldName: string}&array<string, mixed>` is not supported by phpstan so it won't fix anything.

Reply:
This could be the right way ; wdyt ? https://github.com/sonata-project/SonataAdminBundle/pull/7677

Reply:
Yes, indeed, avoid using arrays looks like a good idea, but then it might be for next major. Here we can't do a lot for it, unless we do some magic tricks to be BC